04.ls/ls.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+COLUMN = 3
+SPACE = 1
+
+def main
+  current_path = '../../dummy/' # "#{Dir.getwd}/"
+  current_entries = Dir.entries(current_path).sort
+  hidden_removed_entries = current_entries.delete_if { |entry| entry.start_with?('.') }
+  distinct_entries = distinguish_entries(hidden_removed_entries, current_path)
+  row = (distinct_entries.length.to_f / COLUMN).ceil
+  aligned_entries = align_entries(row, distinct_entries)
+
+  width = calculate_width(distinct_entries)
+  aligned_entries.each do |col_entries|
+    print_col(width, col_entries)
+    puts
+  end
+end
+
+def distinguish_entries(entries, absolute_path)
```

Comment:
```suggestion
def distinguish_entries(entries, path)
```

何か強いこだわりがなければここもpathだけで良いと思います。

あと、distinguishは区別するという意味ですが、「区別してそれからどうするの？」という印象を受けます。
「どうするの？」の方をメソッド名にした方がいいかもしれません。